astr303a.py

```diff
@@ -0,0 +1,571 @@
+import numpy as np
+import pylab as py
+from scipy import optimize
+from matplotlib.patches import Ellipse
+import subprocess
+import sys
+import re
+import os
+import io
+import scipy
+import shutil
+from matplotlib import pylab
+from matplotlib import rc
+from matplotlib import pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+
+#Create a pdf file that all figures will be saved to.
+global pp
+pp = PdfPages('GALAXIES.pdf')
```

Comment:
Likewise, generally you want to make the output filename configurable, put this in a main() function, and then use the plot (pp) object as a parameter to any functions that do plotting.